File: dotnet/GitHub/Models/PullRequest__links.cs
using Microsoft.Kiota.Abstractions.Serialization;
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
namespace GitHubTodoDemo.GitHub.Models {
    public class PullRequest__links : IAdditionalDataHolder, IParsable {
        /// <summary>Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.</summary>
        public IDictionary<string, object> AdditionalData { get; set; }
        /// <summary>Hypermedia Link</summary>
#if NETSTANDARD2_1_OR_GREATER || NETCOREAPP3_1_OR_GREATER
#nullable enable
        public Link? Comments { get; set; }
#nullable restore
#else
        public Link Comments { get; set; }
#endif
        /// <summary>Hypermedia Link</summary>
#if NETSTANDARD2_1_OR_GREATER || NETCOREAPP3_1_OR_GREATER
#nullable enable
        public Link? Commits { get; set; }
#nullable restore
#else
        public Link Commits { get; set; }
#endif
        /// <summary>Hypermedia Link</summary>
#if NETSTANDARD2_1_OR_GREATER || NETCOREAPP3_1_OR_GREATER
#nullable enable
        public Link? Html { get; set; }
#nullable restore
#else
        public Link Html { get; set; }
#endif
        /// <summary>Hypermedia Link</summary>
#if NETSTANDARD2_1_OR_GREATER || NETCOREAPP3_1_OR_GREATER
#nullable enable
        public Link? Issue { get; set; }
#nullable restore
#else
        public Link Issue { get; set; }
#endif
        /// <summary>Hypermedia Link</summary>
#if NETSTANDARD2_1_OR_GREATER || NETCOREAPP3_1_OR_GREATER
#nullable enable
        public Link? Review_comment { get; set; }
#nullable restore
#else
        public Link Review_comment { get; set; }
#endif
        /// <summary>Hypermedia Link</summary>
#if NETSTANDARD2_1_OR_GREATER || NETCOREAPP3_1_OR_GREATER
#nullable enable
        public Link? Review_comments { get; set; }
#nullable restore
#else
        public Link Review_comments { get; set; }
#endif
        /// <summary>Hypermedia Link</summary>
#if NETSTANDARD2_1_OR_GREATER || NETCOREAPP3_1_OR_GREATER
#nullable enable
        public Link? Self { get; set; }
#nullable restore
#else
        public Link Self { get; set; }
#endif
        /// <summary>Hypermedia Link</summary>
#if NETSTANDARD2_1_OR_GREATER || NETCOREAPP3_1_OR_GREATER
#nullable enable
        public Link? Statuses { get; set; }
#nullable restore
#else
        public Link Statuses { get; set; }
#endif
        /// <summary>
        /// Instantiates a new PullRequest__links and sets the default values.
        /// </summary>
        public PullRequest__links() {
            AdditionalData = new Dictionary<string, object>();
        }
        /// <summary>
        /// Creates a new instance of the appropriate class based on discriminator value
        /// </summary>
        /// <param name="parseNode">The parse node to use to read the discriminator value and create the object</param>
        public static PullRequest__links CreateFromDiscriminatorValue(IParseNode parseNode) {
            _ = parseNode ?? throw new ArgumentNullException(nameof(parseNode));
            return new PullRequest__links();
        }
        /// <summary>
        /// The deserialization information for the current model
        /// </summary>
        public IDictionary<string, Action<IParseNode>> GetFieldDeserializers() {
            return new Dictionary<string, Action<IParseNode>> {
                {"comments", n => { Comments = n.GetObjectValue<Link>(Link.CreateFromDiscriminatorValue); } },
                {"commits", n => { Commits = n.GetObjectValue<Link>(Link.CreateFromDiscriminatorValue); } },
                {"html", n => { Html = n.GetObjectValue<Link>(Link.CreateFromDiscriminatorValue); } },
                {"issue", n => { Issue = n.GetObjectValue<Link>(Link.CreateFromDiscriminatorValue); } },
                {"review_comment", n => { Review_comment = n.GetObjectValue<Link>(Link.CreateFromDiscriminatorValue); } },
                {"review_comments", n => { Review_comments = n.GetObjectValue<Link>(Link.CreateFromDiscriminatorValue); } },
                {"self", n => { Self = n.GetObjectValue<Link>(Link.CreateFromDiscriminatorValue); } },
                {"statuses", n => { Statuses = n.GetObjectValue<Link>(Link.CreateFromDiscriminatorValue); } },
            };
        }
        /// <summary>
        /// Serializes information the current object
        /// </summary>
        /// <param name="writer">Serialization writer to use to serialize this model</param>
        public void Serialize(ISerializationWriter writer) {
            _ = writer ?? throw new ArgumentNullException(nameof(writer));
            writer.WriteObjectValue<Link>("comments", Comments);
            writer.WriteObjectValue<Link>("commits", Commits);
            writer.WriteObjectValue<Link>("html", Html);
            writer.WriteObjectValue<Link>("issue", Issue);
            writer.WriteObjectValue<Link>("review_comment", Review_comment);
            writer.WriteObjectValue<Link>("review_comments", Review_comments);
            writer.WriteObjectValue<Link>("self", Self);
            writer.WriteObjectValue<Link>("statuses", Statuses);
            writer.WriteAdditionalData(AdditionalData);
        }
    }
}

File: java/app/src/main/java/githubtododemo/githubclient/models/PullRequestLinks.java
package githubtododemo.githubclient.models;

import com.microsoft.kiota.serialization.Parsable;
import com.microsoft.kiota.serialization.ParseNode;
import com.microsoft.kiota.serialization.SerializationWriter;
import java.util.HashMap;
import java.util.Map;
import java.util.Objects;
public class PullRequestLinks implements Parsable {
    /** Hypermedia Link */
    private Link comments;
    /** Hypermedia Link */
    private Link commits;
    /** Hypermedia Link */
    private Link html;
    /** Hypermedia Link */
    private Link issue;
    /** Hypermedia Link */
    private Link reviewComment;
    /** Hypermedia Link */
    private Link reviewComments;
    /** Hypermedia Link */
    private Link self;
    /** Hypermedia Link */
    private Link statuses;
    /**
     * Creates a new instance of the appropriate class based on discriminator value
     * @param parseNode The parse node to use to read the discriminator value and create the object
     * @return a PullRequestLinks
     */
    @javax.annotation.Nonnull
    public static PullRequestLinks createFromDiscriminatorValue(@javax.annotation.Nonnull final ParseNode parseNode) {
        Objects.requireNonNull(parseNode);
        return new PullRequestLinks();
    }
    /**
     * Gets the comments property value. Hypermedia Link
     * @return a link
     */
    @javax.annotation.Nullable
    public Link getComments() {
        return this.comments;
    }
    /**
     * Gets the commits property value. Hypermedia Link
     * @return a link
     */
    @javax.annotation.Nullable
    public Link getCommits() {
        return this.commits;
    }
    /**
     * The deserialization information for the current model
     * @return a Map<String, java.util.function.Consumer<ParseNode>>
     */
    @javax.annotation.Nonnull
    public Map<String, java.util.function.Consumer<ParseNode>> getFieldDeserializers() {
        final HashMap<String, java.util.function.Consumer<ParseNode>> deserializerMap = new HashMap<String, java.util.function.Consumer<ParseNode>>(8);
        deserializerMap.put("comments", (n) -> { this.setComments(n.getObjectValue(Link::createFromDiscriminatorValue)); });
        deserializerMap.put("commits", (n) -> { this.setCommits(n.getObjectValue(Link::createFromDiscriminatorValue)); });
        deserializerMap.put("html", (n) -> { this.setHtml(n.getObjectValue(Link::createFromDiscriminatorValue)); });
        deserializerMap.put("issue", (n) -> { this.setIssue(n.getObjectValue(Link::createFromDiscriminatorValue)); });
        deserializerMap.put("review_comment", (n) -> { this.setReviewComment(n.getObjectValue(Link::createFromDiscriminatorValue)); });
        deserializerMap.put("review_comments", (n) -> { this.setReviewComments(n.getObjectValue(Link::createFromDiscriminatorValue)); });
        deserializerMap.put("self", (n) -> { this.setSelf(n.getObjectValue(Link::createFromDiscriminatorValue)); });
        deserializerMap.put("statuses", (n) -> { this.setStatuses(n.getObjectValue(Link::createFromDiscriminatorValue)); });
        return deserializerMap;
    }
    /**
     * Gets the html property value. Hypermedia Link
     * @return a link
     */
    @javax.annotation.Nullable
    public Link getHtml() {
        return this.html;
    }
    /**
     * Gets the issue property value. Hypermedia Link
     * @return a link
     */
    @javax.annotation.Nullable
    public Link getIssue() {
        return this.issue;
    }
    /**
     * Gets the review_comment property value. Hypermedia Link
     * @return a link
     */
    @javax.annotation.Nullable
    public Link getReviewComment() {
        return this.reviewComment;
    }
    /**
     * Gets the review_comments property value. Hypermedia Link
     * @return a link
     */
    @javax.annotation.Nullable
    public Link getReviewComments() {
        return this.reviewComments;
    }
    /**
     * Gets the self property value. Hypermedia Link
     * @return a link
     */
    @javax.annotation.Nullable
    public Link getSelf() {
        return this.self;
    }
    /**
     * Gets the statuses property value. Hypermedia Link
     * @return a link
     */
    @javax.annotation.Nullable
    public Link getStatuses() {
        return this.statuses;
    }
    /**
     * Serializes information the current object
     * @param writer Serialization writer to use to serialize this model
     * @return a void
     */
    @javax.annotation.Nonnull
    public void serialize(@javax.annotation.Nonnull final SerializationWriter writer) {
        Objects.requireNonNull(writer);
        writer.writeObjectValue("comments", this.getComments());
        writer.writeObjectValue("commits", this.getCommits());
        writer.writeObjectValue("html", this.getHtml());
        writer.writeObjectValue("issue", this.getIssue());
        writer.writeObjectValue("review_comment", this.getReviewComment());
        writer.writeObjectValue("review_comments", this.getReviewComments());
        writer.writeObjectValue("self", this.getSelf());
        writer.writeObjectValue("statuses", this.getStatuses());
    }
    /**
     * Sets the comments property value. Hypermedia Link
     * @param value Value to set for the comments property.
     * @return a void
     */
    @javax.annotation.Nonnull
    public void setComments(@javax.annotation.Nullable final Link value) {
        this.comments = value;
    }
    /**
     * Sets the commits property value. Hypermedia Link
     * @param value Value to set for the commits property.
     * @return a void
     */
    @javax.annotation.Nonnull
    public void setCommits(@javax.annotation.Nullable final Link value) {
        this.commits = value;
    }
    /**
     * Sets the html property value. Hypermedia Link
     * @param value Value to set for the html property.
     * @return a void
     */
    @javax.annotation.Nonnull
    public void setHtml(@javax.annotation.Nullable final Link value) {
        this.html = value;
    }
    /**
     * Sets the issue property value. Hypermedia Link
     * @param value Value to set for the issue property.
     * @return a void
     */
    @javax.annotation.Nonnull
    public void setIssue(@javax.annotation.Nullable final Link value) {
        this.issue = value;
    }
    /**
     * Sets the review_comment property value. Hypermedia Link
     * @param value Value to set for the reviewComment property.
     * @return a void
     */
    @javax.annotation.Nonnull
    public void setReviewComment(@javax.annotation.Nullable final Link value) {
        this.reviewComment = value;
    }
    /**
     * Sets the review_comments property value. Hypermedia Link
     * @param value Value to set for the reviewComments property.
     * @return a void
     */
    @javax.annotation.Nonnull
    public void setReviewComments(@javax.annotation.Nullable final Link value) {
        this.reviewComments = value;
    }
    /**
     * Sets the self property value. Hypermedia Link
     * @param value Value to set for the self property.
     * @return a void
     */
    @javax.annotation.Nonnull
    public void setSelf(@javax.annotation.Nullable final Link value) {
        this.self = value;
    }
    /**
     * Sets the statuses property value. Hypermedia Link
     * @param value Value to set for the statuses property.
     * @return a void
     */
    @javax.annotation.Nonnull
    public void setStatuses(@javax.annotation.Nullable final Link value) {
        this.statuses = value;
    }
}
